### DISCUSSION TITLE
Indication for fetching deprecated fields

### DISCUSSION THREAD

**Discussion by osher-sade** [graphql-java/graphql-java#2237] [Ideas]
Hello,
Is there any way to know in runtime, if there are any deprecated fields that are being fetched?
If not, consider adding a Boolean field as an indication, maybe as part of `DataFetchingEnvironment`?

**Comment by bbakerman** [comment]
It would not be a common enough question to put a boolean on DFE but it can be easily found

```
boolean deprecated = dataFetchingEnvironment.getFieldDefinition().isDeprecated();
```
